source/dotnet/Library/AdaptiveCards/AdaptiveCard.cs

```diff
@@ -218,5 +218,96 @@ public bool ShouldSerializeJsonSchema()
         {
             return false;
         }
+
+        /// <summary>
+        /// Get resource information for all Images/Media in the whole card
+        /// TODO: Add Media information to the list when Media type is added
+        /// </summary>
+        /// <returns>An array of all card resource information</returns>
+        public RemoteResourceInformation[] GetResourceInformation()
+        {
+            return GetResourceInformationInCard(this).ToArray();
+        }
+
+        private List<RemoteResourceInformation> GetResourceInformationInCard(AdaptiveCard card)
+        {
+            // Initialize the result array
+            List<RemoteResourceInformation> resourceInformationList = new List<RemoteResourceInformation>();
+
+            // Get background image
+            if (!String.IsNullOrEmpty(card.BackgroundImageString))
+            {
+                resourceInformationList.Add(new RemoteResourceInformation(
+                    card.BackgroundImageString,
+                    typeof(AdaptiveImage),
+                    null
+                ));
+            }
+
+            // Get all images in body
+            foreach (AdaptiveElement bodyElement in card.Body)
+            {
+                resourceInformationList.AddRange(GetResourceInformationInElement(bodyElement));
+            }
+
+            // Get all images in ShowCard actions
+            // TODO: Add resource information for iconUrl after it's added
+            foreach (AdaptiveAction action in card.Actions)
+            {
+                if (action is AdaptiveShowCardAction showCardAction)
+                {
+                    resourceInformationList.AddRange(GetResourceInformationInCard(showCardAction.Card));
+                }
+            }
+
+            return resourceInformationList;
+        }
+
+        private List<RemoteResourceInformation> GetResourceInformationInElement(AdaptiveElement element)
+        {
+            List<RemoteResourceInformation> resourceInformationList = new List<RemoteResourceInformation>();
+
+            // Base case
+            if (element is AdaptiveImage imageElement && !String.IsNullOrEmpty(imageElement.UrlString))
+            {
+                resourceInformationList.Add(new RemoteResourceInformation(
+                    imageElement.UrlString,
+                    typeof(AdaptiveImage),
+                    null
+                ));
+            }
```

Comment:
nit: extra space at LN279

Reply:
hm I can't see it

Reply:
sorry i meant new line between if and else if stmt.

---
In reply to: [207416337](https://github.com/Microsoft/AdaptiveCards/pull/1745#discussion_r207416337) [](ancestors = 207416337)